addons/website_crm/controllers/main.py

```diff
@@ -60,6 +60,21 @@ def _handle_website_form(self, model_name, **kwargs):
     def insert_record(self, request, model, values, custom, meta=None):
         is_lead_model = model.model == 'crm.lead'
         if is_lead_model:
+            values_email_normalized = tools.email_normalize(values.get('email_from'))
+            if values_email_normalized:
+                visitor = request.env['website.visitor']._get_visitor_from_request()
+                visitor_partner = visitor.partner_id
+                if visitor and visitor_partner and visitor_partner.email_normalized == values_email_normalized:
+                    # Here, 'phone' in values has already been formatted, see _handle_website_form.
+                    values_phone = values.get('phone')
+                    # We write partner id on crm only if no phone exists on partner or in input,
+                    # or if both (after formating) are the same. This way we get additional phone
+                    # if possible, without modifying an existing one.
+                    if values_phone and visitor_partner.phone:
+                        if visitor_partner._phone_format(visitor_partner.phone) == values_phone:
```

Comment:
no need to provide country and format to be sure to compare the same that in _handle_website_form ?

Reply:
1) Au niveau du format, la méthode _phone_format sur res.partner contient un force_format = 'INTERNATIONAL'. Idem dans _handle_website_form.

2) Pour le pays:
 -VISITOR PARTNER : la méthode _phone_format utilise le pays du partner ou de sa compagnie. ( Tout changement du phone sur le partner passe par cette méthode, donc le format sera adéquat normalement )
 -FORM : le formatting du phone du form est déjà fait à cette étape, en utilisant le pays présent dans le record si il y en a un, et le pays de la session sinon (geoip), dans la méthode _handle_website_form.

->Si il y a une différence, on n'écrit pas le partner sur le lead, évitant toute collision / modification non désirée du partner. En revanche, le phone n'est pas perdu et est bien présent sur le lead.

N'hésite pas si tu as des questions / remarques, je reste dispo :)

Reply:
Pour éviter d'avoir des comportements différents, je peux aussi change la fonction _get_country, utilisée dans _handle_website_form, pour prendre par défaut le pays du partner connecté? En reprenant la même logique que pour partner, ça donnerait:

'''python
def _get_country(self):
        visitor = request.env['website.visitor']._get_visitor_from_request()
        visitor_partner = visitor.partner_id
        if visitor and visitor_partner:
            country = visitor_partner.country_id or request.env.company.country_id
            if country:
                return country
            else:
                country_code = request.session.geoip and request.session.geoip.get('country_code') or False
                if country_code:
                    return request.env['res.country'].sudo().search([('code', '=', country_code)], limit=1)
                return request.env['res.country']

'''